src/kernel/src/cmdline/mod.rs

```diff
@@ -131,24 +131,31 @@ impl Cmdline {
         valid_element(v)?;
         self.has_capacity(k.len() + v.len() + 1)?;
 
-        self.start_push();
-        self.line.push_str(k);
-        self.line.push('=');
-        self.line.push_str(v);
+        if let Some(index) = self.line.find("--") {
+            self.line.insert_str(index, &format!("{}={} ", k, v));
+        } else {
+            self.start_push();
+            self.line.push_str(&format!("{}={}", k, v));
+        }
```

Comment:
In order to make this logic simpler and more generic, could we just rename this method from `insert` to `prepend` and insert these custom strings at the beginning of the cmdline ? Would this work? This way they would always be insreted before `--`.

Reply:
I like this suggestion a lot. Take a look.